e2e/playwright/test-utils.ts

```diff
@@ -1056,3 +1010,59 @@ export async function createProjectAndRenameIt({
 export function executorInputPath(fileName: string): string {
   return join('src', 'wasm-lib', 'tests', 'executor', 'inputs', fileName)
 }
+
+export async function doAndWaitForImageDiff(
+  page: Page,
+  fn: () => Promise<unknown>,
+  diffCount = 200
+) {
+  return new Promise<boolean>((resolve) => {
+    ;(async () => {
+      await page.screenshot({
+        path: './e2e/playwright/temp1.png',
+        fullPage: true,
+      })
+      await fn()
+      const isImageDiff = async () => {
+        await page.screenshot({
+          path: './e2e/playwright/temp2.png',
+          fullPage: true,
+        })
+        const screenshot1 = PNG.sync.read(
+          await fsp.readFile('./e2e/playwright/temp1.png')
+        )
+        const screenshot2 = PNG.sync.read(
+          await fsp.readFile('./e2e/playwright/temp2.png')
+        )
+        const actualDiffCount = pixelMatch(
+          screenshot1.data,
+          screenshot2.data,
+          null,
+          screenshot1.width,
+          screenshot2.height
+        )
+        return actualDiffCount > diffCount
+      }
+
+      // run isImageDiff every 50ms until it returns true or 5 seconds have passed (100 times)
+      let count = 0
+      const interval = setInterval(() => {
+        ;(async () => {
+          count++
+          if (await isImageDiff()) {
+            clearInterval(interval)
+            resolve(true)
+          } else if (count > 100) {
+            clearInterval(interval)
+            resolve(false)
+          }
+        })().catch(reportRejection)
+      }, 50)
+    })().catch(reportRejection)
+  })
+}
```

Comment:
well this is an existing util that was tied to the big `getUtils` object, so I pulled it out so that it can be exported, and left it in `test-utils` for now, not sure if `test-utils.ts` will get deprecated at some point. but I don't think it makes sense for any particular fixture, I think we should avoid putting this in directly into our test, and more so used to make the fixtures more robust, for example here `startSketchPlaneSelection` in `toolbarFixture.ts`
```ts
startSketchPlaneSelection = async () =>
    doAndWaitForImageDiff(this.page, () => this.startSketchBtn.click(), 500)
```
Because we're not ready for the user to select a plane until the default planes appear, and avoids putting `doAndWaitForImageDiff` in the high-level test.